spinedb_api/db_mapping_base.py

```diff
@@ -360,11 +393,13 @@ def _unique_key_value_to_id(self, key, value, fetch=True):
             int
         """
         id_by_unique_value = self._id_by_unique_key_value.get(key, {})
-        if not id_by_unique_value and fetch:
+        value = tuple(tuple(x) if isinstance(x, list) else x for x in value)
+        item_id = id_by_unique_value.get(value)
+        if item_id is None and fetch:
             self._db_map.do_fetch_all(self._item_type)
             id_by_unique_value = self._id_by_unique_key_value.get(key, {})
-        value = tuple(tuple(x) if isinstance(x, list) else x for x in value)
-        return id_by_unique_value.get(value)
+            item_id = id_by_unique_value.get(value)
+        return item_id
 
```

Comment:
Just to be sure - there is no functionality change here, just some optimization to avoid innecessary calls to do_fetch_more... right?

Reply:
Right.